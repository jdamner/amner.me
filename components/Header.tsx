import React from "react";

export default function Header(
  props: React.DetailedHTMLProps<
    React.HTMLProps<HTMLDivElement>,
    HTMLDivElement
  > & { title: string },
) {
  const { title, children, ...rest } = props;

  return (
    <section
      id="header"
      className="bg-black corner-blob text-orange-100 py-4 px-4"
      {...rest}
    >
      <div className="container py-5 mx-auto ">
        <h1 className="font-sans text-5xl sm:text-8xl lg:text-9xl font-black uppercase mb-5 lg:w-1/2">
          {title}
        </h1>
        <div className="text-xl md:text-2xl prose max-w-screen-md">
          {children}
        </div>
      </div>
    </section>
  );
}
