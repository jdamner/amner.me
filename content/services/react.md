---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
