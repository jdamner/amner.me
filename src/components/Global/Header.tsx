import { before, after } from "../../api/Utils";

export default function Title({ subtitle, title, children }: { subtitle?: string, title: string, children?: JSX.Element | JSX.Element[] | string }): JSX.Element {

    const beforeAfterClasses = [
        'content',
        'w-full',
        'bg-slate-900',
        'inline-block',
        'h-3',
    ];

    const subtitleClasses = [
        'font-sans',
        'text-lg',
        'md:text-xl',
        'font-black',
        'uppercase',
        'flex',
        'justify-center',
        'items-center',
        'w-full',
    ];

    return (
        <section id="title" className='bg-slate-200 py-5'>
            <div className='container mx-auto text-center py-5 px-3 md:px-0'>
                {subtitle && <h2 className={subtitleClasses.concat(before(beforeAfterClasses), after(beforeAfterClasses)).join(' ')}><span className='basis-1/2'>{subtitle}</span></h2>}
                <h1 className="font-sans text-4xl md:text-9xl font-black uppercase mb-5">{title}</h1>
                {children}
            </div>
        </section>
    );
}
