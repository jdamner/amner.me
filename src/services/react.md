---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.