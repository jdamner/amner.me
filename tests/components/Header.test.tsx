import React from "react"
import { render } from "@testing-library/react"

import Header from "../../src/components/Global/Header"

describe("Header", () => {
  it("renders correctly", () => {
    const { container } = render(<Header title="Page Not Found" subtitle="Not Found"><p>Something Here</p></Header>)
    expect(container).toMatchSnapshot()
  })
})
