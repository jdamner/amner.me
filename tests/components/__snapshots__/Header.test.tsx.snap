// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <section
    class="bg-slate-200 py-5"
    id="title"
  >
    <div
      class="container mx-auto text-center py-5"
    >
      <h2
        class="font-sans text-xl font-black uppercase flex justify-center items-center w-full before:content before:w-full before:bg-slate-900 before:inline-block before:h-4 after:content after:w-full after:bg-slate-900 after:inline-block after:h-4"
      >
        <span
          class="basis-1/2"
        >
          Not Found
        </span>
      </h2>
      <h1
        class="font-sans text-9xl font-black uppercase mb-5"
      >
        Page Not Found
      </h1>
      <p>
        Something Here
      </p>
    </div>
  </section>
</div>
`;
